### ISSUE TITLE
Warnings &quot;__VA_ARGS__ is a GNU extension&quot; in clang-3.7 compiler

### ISSUE THREAD

**Issue by pawelsalawa** [pawelsalawa/sqlitestudio#2984]
_(This issue was migrated from the old bug tracker of SQLiteStudio)_

Original ID from old bug tracker: 3100
Originally created at: Thu Jun 16 21:35:41 2016
Originally last updated at: Thu Jun 16 21:35:41 2016

Seeing a lot of these warnings:
./config\\_builder.h:14:90: warning: token pasting of '','' and \\_\\_VA\\_ARGS\\_\\_ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]\n\n**Operating system:**\nFreeBSD 64-bit\n\n**Version:**\n3.1.0

**Comment by pawelsalawa** [comment]
Originally created at: Thu Jun 16 21:35:41 2016

Yes, SQLiteStudio sourcecode uses variadic macros. For now I don't know a standarized way to implement features I needed, so for now they have to stay.

**Comment by pawelsalawa** [comment]
Originally created at: Thu Jun 16 21:35:41 2016

Clang says that to silence the warning -Wgnu-zero-variadic-macro-arguments needs to be added, I think.

**Comment by pawelsalawa** [comment]
Originally created at: Thu Jun 16 21:35:41 2016

It's a bit more complicated. I think they have answer here <a target="\\_blank" href="http://stackoverflow.com/questions/26932749/how-can-wgnu-zero-variadic-macro-arguments-warning-be-turned-off-with-clang">http://stackoverflow.com/questions/26932749/how-can-wgnu-zero-variadic-macro-arguments-warning-be-turned-off-with-clang</a>

I will try it out when I have a little time for it.

**Comment by pawelsalawa** [comment]
Originally created at: Thu Jun 16 21:35:41 2016

_Status changed to **assigned**_

**Comment by landryb** [comment]
fwiw, the warning is still shown with clang 13 on OpenBSD:
```
../../../SQLiteStudio3/coreSQLiteStudio/config_builder.h:14:90: warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
```

**Comment by pawelsalawa** [comment]
Yes. The issue is still open, as there are more important things than getting rid of a warning, which actually is not that simple as it may look like.
